commit 1fbfea71fb42b2e79f9e342c1843bd26a8703536
Author: q66 <q66@chimera-linux.org>
Date:   Thu Aug 25 01:20:58 2022 +0200

    An all-in-one patch that fixes several issues:
    
    1) UnscaledCycleClock not fully implemented for ppc*-musl (disabled on musl)
    2) powerpc stacktrace implementation only works on glibc (disabled on musl)
    3) powerpc stacktrace implementation has ppc64 assumptions (fixed)
    4) examine_stack.cpp makes glibc assumptions on powerpc (fixed)

diff --git a/deps/v8/third_party/abseil-cpp/absl/base/internal/unscaledcycleclock.cc b/deps/v8/third_party/abseil-cpp/absl/base/internal/unscaledcycleclock.cc
index a0bf3a6..103b4f6 100644
--- a/deps/v8/third_party/abseil-cpp/absl/base/internal/unscaledcycleclock.cc
+++ b/deps/v8/third_party/abseil-cpp/absl/base/internal/unscaledcycleclock.cc
@@ -20,7 +20,7 @@
 #include <intrin.h>
 #endif
 
-#if defined(__powerpc__) || defined(__ppc__)
+#if (defined(__powerpc__) || defined(__ppc__)) && defined(__GLIBC__)
 #ifdef __GLIBC__
 #include <sys/platform/ppc.h>
 #elif defined(__FreeBSD__)
@@ -58,7 +58,7 @@ double UnscaledCycleClock::Frequency() {
   return base_internal::NominalCPUFrequency();
 }
 
-#elif defined(__powerpc__) || defined(__ppc__)
+#elif (defined(__powerpc__) || defined(__ppc__)) && defined(__GLIBC__)
 
 int64_t UnscaledCycleClock::Now() {
 #ifdef __GLIBC__
diff --git a/deps/v8/third_party/abseil-cpp/absl/base/internal/unscaledcycleclock_config.h b/deps/v8/third_party/abseil-cpp/absl/base/internal/unscaledcycleclock_config.h
index 43a3dab..da63c74 100644
--- a/deps/v8/third_party/abseil-cpp/absl/base/internal/unscaledcycleclock_config.h
+++ b/deps/v8/third_party/abseil-cpp/absl/base/internal/unscaledcycleclock_config.h
@@ -20,9 +20,9 @@
 #endif
 
 // The following platforms have an implementation of a hardware counter.
-#if defined(__i386__) || defined(__x86_64__) || defined(__aarch64__) || \
-    defined(__powerpc__) || defined(__ppc__) || defined(_M_IX86) ||     \
-    (defined(_M_X64) && !defined(_M_ARM64EC))
+#if defined(__i386__) || defined(__x86_64__) || defined(__aarch64__) ||        \
+    ((defined(__powerpc__) || defined(__ppc__)) && defined(__GLIBC)) ||        \
+    defined(_M_IX86) || (defined(_M_X64) && !defined(_M_ARM64EC))
 #define ABSL_HAVE_UNSCALED_CYCLECLOCK_IMPLEMENTATION 1
 #else
 #define ABSL_HAVE_UNSCALED_CYCLECLOCK_IMPLEMENTATION 0
diff --git a/deps/v8/third_party/abseil-cpp/absl/debugging/internal/examine_stack.cc b/deps/v8/third_party/abseil-cpp/absl/debugging/internal/examine_stack.cc
index 3dd6ba1..ec596c7 100644
--- a/deps/v8/third_party/abseil-cpp/absl/debugging/internal/examine_stack.cc
+++ b/deps/v8/third_party/abseil-cpp/absl/debugging/internal/examine_stack.cc
@@ -177,8 +177,10 @@ void* GetProgramCounter(void* const vuc) {
     return reinterpret_cast<void*>(context->uc_mcontext.pc);
 #elif defined(__powerpc64__)
     return reinterpret_cast<void*>(context->uc_mcontext.gp_regs[32]);
-#elif defined(__powerpc__)
+#elif defined(__powerpc__) && defined(__GLIBC__)
     return reinterpret_cast<void*>(context->uc_mcontext.uc_regs->gregs[32]);
+#elif defined(__powerpc__)
+    return reinterpret_cast<void*>(context->uc_mcontext.gregs[32]);
 #elif defined(__riscv)
     return reinterpret_cast<void*>(context->uc_mcontext.__gregs[REG_PC]);
 #elif defined(__s390__) && !defined(__s390x__)
diff --git a/deps/v8/third_party/abseil-cpp/absl/debugging/internal/stacktrace_config.h b/deps/v8/third_party/abseil-cpp/absl/debugging/internal/stacktrace_config.h
index 3929b1b..23d5e50 100644
--- a/deps/v8/third_party/abseil-cpp/absl/debugging/internal/stacktrace_config.h
+++ b/deps/v8/third_party/abseil-cpp/absl/debugging/internal/stacktrace_config.h
@@ -60,7 +60,7 @@
 #elif defined(__i386__) || defined(__x86_64__)
 #define ABSL_STACKTRACE_INL_HEADER \
   "absl/debugging/internal/stacktrace_x86-inl.inc"
-#elif defined(__ppc__) || defined(__PPC__)
+#elif (defined(__ppc__) || defined(__PPC__)) && defined(__GLIBC__)
 #define ABSL_STACKTRACE_INL_HEADER \
   "absl/debugging/internal/stacktrace_powerpc-inl.inc"
 #elif defined(__aarch64__)
